### PR TITLE
Pigeon example

### PR DESCRIPTION
**Pull request by cbisaillon**

### CODE REVIEW

**Comment by Kansuler** [comment]
Haha, so this is the idea you got in Twitter! I like it.

Which gave me a thought, can a pigeon fly without legs? 🤔

File: toys/pigeon/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        }
+
+        // Try to fly. The pigeon must be fully assembled !
+        pub fn fly(&self) {
+            assert!(self.head_vault.amount() == 1.into(), "Missing head !");
+            assert!(self.body_vault.amount() == 1.into(), "Missing body !");
+            assert!(self.leg_vault.amount() == 2.into(), "Missing legs !");
+            assert!(self.wing_vault.amount() == 2.into(), "Missing wings !");
+
+            info!("The pigeon is flying sucessfuly !")
```
Comment (Kansuler) [inline toys/pigeon/src/lib.rs:63]: Nitpick on spelling of successfully

Comment (cbisaillon) [reply toys/pigeon/src/lib.rs:63]: Thank you !

**Comment by cbisaillon** [comment]
> So this is the idea you got! I like it.
> 
> Which gave me a thought, can a pigeon fly without legs? 🤔

Yes but they can't land: https://www.quora.com/Will-a-bird-be-able-to-fly-if-its-legs-are-broken

**Comment by iamyulong** [comment]
This is a fun example.

**Pull request merged**